addons/account/models/account_payment.py

```diff
@@ -363,22 +363,24 @@ def _onchange_amount(self):
     @api.one
     @api.depends('invoice_ids', 'payment_type', 'partner_type', 'partner_id')
     def _compute_destination_account_id(self):
+        company_ctx = {'force_company': self.company_id.id}
         if self.invoice_ids:
             self.destination_account_id = self.invoice_ids[0].account_id.id
         elif self.payment_type == 'transfer':
             if not self.company_id.transfer_account_id.id:
                 raise UserError(_('Transfer account not defined on the company.'))
             self.destination_account_id = self.company_id.transfer_account_id.id
         elif self.partner_id:
+            partner = self.partner_id.with_context(**company_ctx)
             if self.partner_type == 'customer':
-                self.destination_account_id = self.partner_id.property_account_receivable_id.id
+                self.destination_account_id = partner.property_account_receivable_id.id
             else:
-                self.destination_account_id = self.partner_id.property_account_payable_id.id
+                self.destination_account_id = partner.property_account_payable_id.id
         elif self.partner_type == 'customer':
-            default_account = self.env['ir.property'].get('property_account_receivable_id', 'res.partner')
+            default_account = self.env['ir.property'].with_context(**company_ctx).get('property_account_receivable_id', 'res.partner')
             self.destination_account_id = default_account.id
         elif self.partner_type == 'supplier':
-            default_account = self.env['ir.property'].get('property_account_payable_id', 'res.partner')
+            default_account = self.env['ir.property'].with_context(**company_ctx).get('property_account_payable_id', 'res.partner')
```

Comment:
Don't erase the whole context. Use kwargs instead: `with_context(force_company=self.company_id.id)`

Reply:
This is what the __**__ is for, it expand dict to kwargs! But okay I'll make it explicit

Reply:
@smetl I'm updated the fix, could you re-review it ? Thanks a lot!